File: 05/main.cs


using System.IO.Compression;

var rawStr = "For fear of losing you,I would never cry.For fear of losing you,I would never cry.For fear of losing you,I would never cry.";
var bytes = Encoding.UTF8.GetBytes(rawStr);
Console.WriteLine(bytes.Length);
var ms = new MemoryStream();
using (Stream ds = new DeflateStream(ms, CompressionMode.Compress))
{
    ds.Write(bytes, 0, bytes.Length);
}

byte[] compressed = ms.ToArray();
Console.WriteLine(compressed.Length);

//Decompress
ms = new MemoryStream(compressed);
var primary = new MemoryStream();
using (Stream ds = new DeflateStream(ms, CompressionMode.Decompress))
{
    ds.CopyTo(primary);
}
Console.WriteLine(primary.ToArray().Length);


//private static void CompressFile()
//{
//    using FileStream originalFileStream = File.Open(OriginalFileName, FileMode.Open);
//    using FileStream compressedFileStream = File.Create(CompressedFileName);
//    using var compressor = new GZipStream(compressedFileStream, CompressionMode.Compress);
//    originalFileStream.CopyTo(compressor);
//}

//private static void DecompressFile()
//{
//    using FileStream compressedFileStream = File.Open(CompressedFileName, FileMode.Open);
//    using FileStream outputFileStream = File.Create(DecompressedFileName);
//    using var decompressor = new GZipStream(compressedFileStream, CompressionMode.Decompress);
//    decompressor.CopyTo(outputFileStream);
//}

File: 05/main.js
const zlib = require('zlib')
const rawStr = "For fear of losing you,I would never cry.";
//压缩
zlib.deflate(rawStr, (err, buf) => {
  if (!err) {
    console.log(buf.toString())

    //解压缩
    zlib.inflate(buf, (err, primaryBuf) => {
      if (!err) {
        console.log(primaryBuf.toString());
      }
    })
  }
})

var gzip = zlib.createGzip();
var fs = require("fs");
var inp = fs.createReadStream("input.txt");
var out = fs.createWriteStream("input.txt.gz");

inp.pipe(gzip).pipe(out);
